fe/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java

```diff
@@ -0,0 +1,200 @@
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.clone;
+
+import org.apache.doris.analysis.AddPartitionClause;
+import org.apache.doris.analysis.DistributionDesc;
+import org.apache.doris.analysis.HashDistributionDesc;
+import org.apache.doris.analysis.PartitionKeyDesc;
+import org.apache.doris.analysis.PartitionValue;
+import org.apache.doris.analysis.SingleRangePartitionDesc;
+import org.apache.doris.catalog.Catalog;
+import org.apache.doris.catalog.Column;
+import org.apache.doris.catalog.Database;
+import org.apache.doris.catalog.HashDistributionInfo;
+import org.apache.doris.catalog.OlapTable;
+import org.apache.doris.catalog.RangePartitionInfo;
+import org.apache.doris.catalog.Table;
+import org.apache.doris.catalog.TableProperty;
+import org.apache.doris.common.DdlException;
+import org.apache.doris.common.Pair;
+import org.apache.doris.common.util.DynamicPartitionUtil;
+import org.apache.doris.common.util.MasterDaemon;
+import org.apache.doris.common.util.TimeUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class DynamicPartitionScheduler extends MasterDaemon {
+    private static final Logger LOG = LogManager.getLogger(DynamicPartitionScheduler.class);
+
+    private static final String defaultValue = "N/A";
+    private static String lastSchedulerTime = defaultValue;
+    private static String lastUpdateTime = defaultValue;
+    private static State dynamicPartitionState = State.NORMAL;
+    private static String Msg = defaultValue;
+
+    public enum State {
+        NORMAL,
+        ERROR
+    }
+
+    public static String getLastSchedulerTime() {
+        return lastSchedulerTime;
+    }
+
+    public static void setLastSchedulerTime(String lastSchedulerTime) {
+        DynamicPartitionScheduler.lastSchedulerTime = lastSchedulerTime;
+    }
+
+    public static String getLastUpdateTime() {
+        return lastUpdateTime;
+    }
+
+    public static void setLastUpdateTime(String lastUpdateTime) {
+        DynamicPartitionScheduler.lastUpdateTime = lastUpdateTime;
+    }
+
+    public static State getDynamicPartitionState() {
+        return dynamicPartitionState;
+    }
+
+    public static void setDynamicPartitionState(State state) {
```

Comment:
why using `static` method?. So does all other static methods.

`DynamicPartitionScheduler` is not some kind of "tool" class, like 'Utils.java'. So I think non-static method is better.

And I can't find other caller call this method outside the class, so maybe `private` method is enough, or even remove this method, just set the state using `dynamicPartitionState = xxx;` where you like

Reply:
Done